File: src/main/java/yuensik_cheung/capstone/model/ResizeDragElement.java
package yuensik_cheung.capstone.model;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
/*
 * This is a React component that going to display on the builder page. There attribute are essential for calculating
 * the aircraft perform later the project
 */
@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor

public class ResizeDragElement {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="resizeDragElement_id")
	private Long id;
    @Column(name="project_id")
    private String projectId;
	private String data_x;
	private String data_y;
	private String height;
	private String width;
	private String z_index;
	private String index_num;
	private String src;
	private String view;
	private String className;
	private String tabIndex;
	private String part_type;
	
}

File: app/src/Components/ResizeDragElement/ResizeDragElement.js
import React from 'react'
import styles from './ResizeDragElement.module.css'
import '../BuildWindow/CanvasControl';

const ResizeDragElement = ({ component }) => {
  console.log(component);
  var { src, tabIndex, data_x, data_y, height, width, z_index ,part_type} = component;
  return (
    <img part-type={part_type}  data-x={data_x} data-y={data_y} style={{ zIndex: z_index, height: height, width: width,transform: `translate(${data_x}px, ${data_y}px)` }}tabIndex={tabIndex} src={src} classname2 = "resize_drag_element" className={styles.resize_drag_element + " resize_drag_element"} id={'resize_drag_element_'+tabIndex}/>
  )
}

export default ResizeDragElement